build_package.hs

```diff
@@ -0,0 +1,232 @@
+--      |- licenses/
+--
+-- This script must be run from its containing directory.
+--
+-- TODO Make it robust against when running from other directories.
+
+{-# LANGUAGE BangPatterns           #-}
+{-# LANGUAGE ConstraintKinds        #-}
+{-# LANGUAGE DataKinds              #-}
+{-# LANGUAGE LambdaCase             #-}
+{-# LANGUAGE MultiWayIf             #-}
+{-# LANGUAGE NegativeLiterals       #-}
+{-# LANGUAGE NoImplicitPrelude      #-}
+{-# LANGUAGE OverloadedLabels       #-}
+{-# LANGUAGE OverloadedStrings      #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE StandaloneDeriving     #-}
+{-# LANGUAGE TemplateHaskell        #-}
+{-# LANGUAGE TupleSections          #-}
+{-# LANGUAGE TypeApplications       #-}
+
+module Main where
+
+import Prelude hiding (FilePath)
+
+import qualified Data.Either         as Either
+import qualified Data.List           as List
+import qualified Data.Text           as Text
+import qualified Data.Text.IO        as Text
+import qualified Options.Applicative as Options
+import qualified Path                as Path
+import qualified System.Directory    as Directory
+import qualified System.Environment  as Environment
+import qualified System.FilePath     as FilePath
+import qualified Turtle              as Turtle
+import qualified Turtle.Pattern      as Turtle
+
+import Control.Lens           (makeLenses, (^.))
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad          (when, unless)
+import Data.Text              (Text)
+import Options.Applicative    ((<**>))
+import Path                   (Path, Rel, File, Abs, Dir)
+import System.FilePath        (FilePath, (</>))
+import Turtle                 ((<|>))
+
+
+
+-----------------------
+-- === Constants === --
+-----------------------
+
+projectYaml :: Path Rel File
+projectYaml = $(Path.mkRelFile "./stack.yaml")
+
+releaseOpts :: [Text]
+releaseOpts = ["-fno-omit-interface-pragmas"]
+
+pkgBaseDirFromLocalBin :: FilePath
+pkgBaseDirFromLocalBin = "../../"
+
+stdlibSrcFolderLoc :: Turtle.FilePath
+stdlibSrcFolderLoc = "stdlib/Std/"
+
+licenseSrcFolderLoc :: Turtle.FilePath
+licenseSrcFolderLoc = "package/data/licenses"
+
+dataDistLoc :: Turtle.FilePath
+dataDistLoc = "config/env/"
+
+
+
+-------------------------
+-- === CLI Options === --
+-------------------------
+
+-- === Definition === --
+
+data CommandOpts = CommandOpts
+    { _releaseMode :: Bool
+    , _verbose     :: Bool
+    , _check       :: Bool
+    , _cleanStack  :: Bool
+    } deriving (Eq, Show)
+makeLenses ''CommandOpts
+
+
+-- === API === --
+
+optionsParser :: Options.Parser CommandOpts
+optionsParser = CommandOpts
+    <$> Options.switch (Options.long "release"
+        <> Options.help "Build the Luna package in release mode")
+    <*> Options.switch (Options.long "verbose"
+        <> Options.help "Enable verbose mode while building.")
+    <*> Options.switch (Options.long "check"
+        <> Options.help "Check that the build script itself builds.")
```

Comment:
This intentionally creates a code-path that does nothing to ensure that the script builds using `stack script`. Otherwise every call to this script would inherently have to build the package in some form.

Reply:
So this is more like a workaround of not being able to just make call "stack build" on the script itself. (the "check" happens in stack tool, before our script is run)
Let it be for know, though I think this flag should go away when/if script is promoted to package.

Reply:
It's primarily for use on CI, and is what caught the turtle issue on windows.

Reply:
Yes, certainly better than not having it!